Cry 12/api endpoint metricstimeseries lien duckdb x parquet

vfargue opened the pull request:
[FEAT] Ajout de l'endpoint /metrics/timeseries

Modification du Dockerfile pour installation de duckdb en automatique.
Modification du main.py pour lecture de la route.
Modification de timeseries pour création de la route.

AlexandruPanta committed:
feat init duckdb with data

VersionF committed:
Merge branch 'cry-17-api-init-duckdb-lecture-parquet-table-latest' of github.com:Zypheone94/crypto-viz into CRY-12/api-endpoint-metricstimeseries-lien-duckdb-x-parquet

VersionF committed:
Merge branch 'dev' of github.com:Zypheone94/crypto-viz into CRY-12/api-endpoint-metricstimeseries-lien-duckdb-x-parquet

VersionF committed:
[FEAT] Ajout du Endpoint /metrics/timeseries

VersionF committed:
Merge branch 'dev' of github.com:Zypheone94/crypto-viz into CRY-12/api-endpoint-metricstimeseries-lien-duckdb-x-parquet

VersionF committed:
[FEAT] Ajout de l'endpoint /metrics/timeseries

LombardMaxime commented on ingestor/scraper/Dockerfile:
```diff
@@ -1,11 +1,17 @@
+COPY ingestor/pyproject.toml ingestor/poetry.lock* ./ingestor/
 
-RUN pip install -e .
+RUN pip install --upgrade pip \
```
ceci est un mauvaise pratique
il vaut mieux ajouter les dépenses nécessaire (duckdb) dans le fichier toml et lancer la commande d'installation global (pip install -e)

LombardMaxime commented on ingestor/scraper/Dockerfile:
```diff
@@ -1,11 +1,17 @@
+
+EXPOSE 8080
+
+CMD ["uvicorn", "ingestor.scraper.main:app", "--host", "0.0.0.0", "--port", "8080"]
```
il n'est pas nécessaire de mettre cette commande, la commande dans le docker compose suffit

LombardMaxime commented on ingestor/scraper/Dockerfile:
```diff
@@ -1,11 +1,17 @@
\ No newline at end of file
+COPY ingestor /app/ingestor
+
+EXPOSE 8080
```
il est préférable de faire un .env pour exposer un port (et de l'exposer dans le docker compose que dans le docker file)

LombardMaxime commented on ingestor/scraper/api/routes/metrics.py:
je propose l'idée mais ton modèle à l'air d'être un fichier pour un endpoint, à l'avenir si l'on a plusieurs endpoint faisant des actions "similaire" ne serait-il pas plus pertinant de faire un file "par catégorie" d'endpoint plutôt qu'un file par endpoint ?

LombardMaxime reviewed (changes requested):
Changment à faire obligatoirement sur le dockerfile, et ceux conseillé par copilot (si tu n'est pas sur de comprendre les coms concernant le dockerfile met moi un message si tu as besoin d'aide / dis moi et je m'en chargerais après que tu ai update les coms copilot)

VersionF committed:
[FIX] Fix des remontées de MR

Modification du Dockerfile
Modification du pyproject.toml
Renommage de l'endpoint timeseries en metrics et modification du fichiers pour correspondre à ce changement
